contrib/grain_size/equilibrium_grain_size_pinned_state.ipynb

```diff
@@ -0,0 +1,167 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 189,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 229,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# define constants here\n",
+    "q                       = 4.\n",
+    "grain_growth_prefactor  = 1.6e-22\n",
+    "grain_growth_activation = 39072\n",
+    "f1                      = 1e-6\n",
+    "phase_distribution      = 0.6*0.4\n",
+    "\n",
+    "hg         = (2/np.pi)*(2/np.pi)\n",
+    "R_times_T  = 8.314*1600"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\\begin{equation}\n",
+    "0 = \\frac {3\\eta_g\\sqrt hg^{-q}}{(q R^{q-1})} \\exp(\\frac{-Eg}{R_gT})  - \\frac{f hg \\Psi}{(3\\eta)} R^2\n",
+    "\\end{equation}\n",
+    "\n",
+    "Here, $\\eta_g$ is phase distribution function and $\\sqrt(hg) = 2/\\pi$. Combining the prefactors of R (grain_size) into one such that the above equation can be written as :\n",
+    "\n",
+    "\\begin{equation}\n",
+    "0 = \\frac{\\text{grain_growth_term}}{R^{(q-1)}} - \\text{grain_reduction_term}*R^2\n",
+    "\\end{equation}\n",
+    "\n",
+    "For q = 4:\n",
+    "\\begin{equation}\n",
+    "R = \\left(\\frac{\\text{grain_growth_term}}{\\text{grain_reduction_term}}\\right)^{1/5}\n",
+    "\\end{equation}\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 230,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def equilibrium_grain_size (viscosity, q):\n",
+    "    \n",
+    "    shear_heating = 4*strain_rate*strain_rate*viscosity\n",
+    "    \n",
+    "    grain_reduction_term = f1*hg*shear_heating/(3*phase_distribution)\n",
+    "    \n",
+    "    grain_growth_term    = (hg**(-q/2) * 3 * phase_distribution * grain_growth_prefactor * \\\n",
+    "                                 np.exp(-grain_growth_activation/R_times_T))/q\n",
+    "    \n",
+    "    return (grain_growth_term/grain_reduction_term)**(1/(q+1))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\\begin{equation}\n",
+    "\\eta = \\frac{1}{2}*A^{\\frac{-1}{n}}*\\dot{\\eta}^{1/n - 1} * \\exp{\\frac{-E_d}{nRT}}\n",
+    "\\end{equation}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 231,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# for the constant strain rate (10^-5/yr) test case and dislocation creep only\n",
+    "strain_rate            = (1e-5/(3600*24*365.25))/2\n",
```

Comment:
I don't understand why the strain rate is divided by two here. Can you explain that in a comment?

Reply:
I had the incorrect comment earlier, the velocity is 10^5m/year and we compute strain rate using that. I added the comment in the last commit, let me know if it doesn't make sense.

Reply:
Yes, that makes sense!